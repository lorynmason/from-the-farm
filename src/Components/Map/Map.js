import React, { Component } from 'react';
import accessToken from '../../mapbox-accessToken.js';
import L from 'leaflet';

export class Map extends Component {

  componentDidMount() {
    this.createMap();
  }

  createMap = () => {
    const map = L.map('map').setView([39.750614, -104.996775], 9);
    const { coordinates } = this.props;

    coordinates.forEach((location) => {
      console.log(location)
      const marker = L.marker(location).addTo(map);
      marker.bindPopup(`<h4>${location}</h4>`)
    });

    L.tileLayer('https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}', {
      attribution: 'Map data &copy; <a href="https://www.openstreetmap.org/">OpenStreetMap</a> contributors, <a href="https://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>, Imagery Â© <a href="https://www.mapbox.com/">Mapbox</a>',
      maxZoom: 18,
      id: 'mapbox.streets',
      accessToken
    }).addTo(map);
  }

  render() {
    const dimensions = {
      height: '400px',
      width: '400px'
    }
    return (
      <section id="map" style={dimensions}></section>
    )
  }
}

export default Map;