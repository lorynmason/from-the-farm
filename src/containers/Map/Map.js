import React from 'react'
import { connect } from 'react-redux';
import { Map, Marker, Popup, TileLayer } from 'react-leaflet';
import L from 'leaflet';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export const Mapp = ({ vendors, vendorSearchResults }) => {
  const position = [39.750614, -104.996775]
  let vendorsToShow = vendors
  if(vendorSearchResults.length) {
    vendorsToShow = vendors.filter( vendor => {
      return vendorSearchResults.includes(vendor.id) 
    })
  }
  const customMarker = L.icon({ iconUrl:('https://www.svgrepo.com/show/7496/carrot.svg'), })
  const markers = vendorsToShow.map((vendor) => (
    <Marker position={[vendor.lat, vendor.long]} icon={customMarker} key={vendor.id}>
      <Popup>
        <h3><Link to={`/profile/${vendor.id}`} >{vendor.name}</Link></h3>
        <p>{vendor.address}</p>
        <p>{vendor.phone}</p>
        <p>{vendor.email}</p>
      </Popup>
    </Marker>
  ));

  return (
    <Map center={position} zoom={11}>
      <TileLayer
        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        attribution="&copy; <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors"
      />
      {markers}
    </Map>
  )
}

const mapStateToProps = (state) => ({
  vendors: state.vendors,
  vendorSearchResults: state.vendorSearchResults
});

export default connect(mapStateToProps)(Mapp);

Mapp.propTypes = {
  vendors: PropTypes.array,
  vendorSearchResults: PropTypes.array
}
