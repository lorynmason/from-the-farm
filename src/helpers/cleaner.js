const formatPhoneNumber = (num) => {
  const phoneRegex = /^\(?([0-9]{3})\)?[-. ]?([0-9]{3})[-. ]?([0-9]{4})$/;
  return num.toString().replace(phoneRegex, "($1) $2-$3")
}

export const cleanProducts = (vendors) => {
  return vendors.reduce((productArr, vendor) => {
    const { name, products } = vendor.attributes;
    products.reduce((obj, product) => {
      const itemName = Object.keys(product)[0];

      obj = {
        vendorName: name,
        name: Object.keys(product)[0],
        ...product[itemName]
      }
      productArr.push(obj);
      return obj
    }, {})

    return productArr;
  }, [])
}

export const cleanVendors = (vendors) => {
  return vendors.map( vendor => {
    const { name, address, state, city, phone, email, lat, long, bio } = vendor.attributes
    const formattedNum = formatPhoneNumber(phone)
    return {
      name,
      bio,
      address,
      state, 
      city,
      phone: formattedNum,
      email,
      lat,
      long, 
      id: parseInt(vendor.id)
    }
  })
}

export const cleanUser = (data, token) => {
  const products = cleanUserProducts(data.attributes.products)
  const { name, email, address, phone, bio, lat, long, city, state, zip, img_url } = data.attributes
  return {
    name, 
    email, 
    address, 
    phone: formatPhoneNumber(phone), 
    bio, 
    lat, 
    long, 
    city, 
    state, 
    zip, 
    img_url,
    products, 
    token
  }
}

export const cleanUserProducts = (products) => {
  const productArr = []
  products.reduce((obj, product) => {
    const itemName = Object.keys(product)[0];
    obj = {
      name: Object.keys(product)[0],
      ...product[itemName]
    }
    productArr.push(obj);
    return obj
  }, {})
  return productArr
}

export const cleanItems = (products) => {
  const productNames = products.reduce((arr, product) => {
    if(!arr.includes(product.name)) {
      arr.push(product.name)
    }
    return arr     
  },[]);

  const productIds = products.reduce((arr, product) => {
    if(!arr.includes(product.item_id)) {
      arr.push(product.item_id)
    }
    return arr     
  },[]);

  return productNames.map((name, i) => {
    return {
      name,
      id: productIds[i]
    }
  });
}