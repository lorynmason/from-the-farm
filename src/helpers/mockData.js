export const mockVendors = [
  {
    "id": "5",
    "type": "user",
    "attributes": {
        "name": "Basil Connection",
        "account_type": "vendor",
        "address": "4950 Beach Court",
        "city": "denver",
        "state": "CO",
        "zip": 80221,
        "email": "helter_skelter@example.com",
        "phone": 2313414141,
        "lat": 39.7861784,
        "long": -105.0178452,
        "bio": "The WORLDS Freshest Basil",
        "img_url": null,
        "products": [
            {
                "Berries": {
                    "id": 3,
                    "user_id": 5,
                    "item_id": 1,
                    "price": 1400,
                    "unit": "lb",
                    "description": "lb of berries"
                }
            },
            {
                "Potatoes": {
                    "id": 17,
                    "user_id": 5,
                    "item_id": 4,
                    "price": 170,
                    "unit": "lb",
                    "description": "a lb of potatoes"
                }
            }]
          }
  }
]

export const mockUser = {
  "id": "5",
  "type": "user",
  "attributes": {
    "name": "Basil Connection",
    "account_type": "vendor",
    "address": "4950 Beach Court",
    "city": "denver",
    "state": "CO",
    "zip": 80221,
    "email": "helter_skelter@example.com",
    "phone": 2313414141,
    "lat": 39.7861784,
    "long": -105.0178452,
    "bio": "The WORLDS Freshest Basil",
    "img_url": null,
    "products": [
      {
        "Berries": {
            "id": 3,
            "user_id": 5,
            "item_id": 1,
            "price": 1400,
            "unit": "lb",
            "description": "lb of berries"
        }
      },
      {
        "Potatoes": {
            "id": 17,
            "user_id": 5,
            "item_id": 4,
            "price": 170,
            "unit": "lb",
            "description": "a lb of potatoes"
        }
      }
    ]
  }
}
