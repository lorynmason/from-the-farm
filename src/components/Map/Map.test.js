import React from 'react';
import { Mapp } from './Map';
import { shallow } from 'enzyme';
import { mockVendors } from '../../helpers/mockData'

describe('Map', () => {
  it('should match snapshot', () => {
    let wrapper = shallow(<Mapp vendorSearchResults={[]} vendors={mockVendors}/>)
    expect(wrapper).toMatchSnapshot()
  });
  it('should match snapshot', () => {
    let wrapper = shallow(<Mapp vendorSearchResults={mockVendors} vendors={mockVendors}/>)
    expect(wrapper).toMatchSnapshot()
  });
});
