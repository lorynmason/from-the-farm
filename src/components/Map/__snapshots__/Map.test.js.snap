// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Mapp
    vendorSearchResults={Array []}
    vendors={
      Array [
        Object {
          "attributes": Object {
            "account_type": "vendor",
            "address": "4950 Beach Court",
            "bio": "The WORLDS Freshest Basil",
            "city": "denver",
            "email": "helter_skelter@example.com",
            "img_url": null,
            "lat": 39.7861784,
            "long": -105.0178452,
            "name": "Basil Connection",
            "phone": 2313414141,
            "products": Array [
              Object {
                "Berries": Object {
                  "description": "lb of berries",
                  "id": 3,
                  "item_id": 1,
                  "price": 1400,
                  "unit": "lb",
                  "user_id": 5,
                },
              },
              Object {
                "Potatoes": Object {
                  "description": "a lb of potatoes",
                  "id": 17,
                  "item_id": 4,
                  "price": 170,
                  "unit": "lb",
                  "user_id": 5,
                },
              },
            ],
            "state": "CO",
            "zip": 80221,
          },
          "id": "5",
          "type": "user",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "center": Array [
        39.750614,
        -104.996775,
      ],
      "children": Array [
        <ForwardRef(Leaflet(TileLayer))
          attribution="© <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors"
          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        />,
        Array [
          <ForwardRef(Leaflet(Marker))
            icon={
              NewClass {
                "_initHooksCalled": true,
                "options": Object {
                  "iconUrl": "https://www.svgrepo.com/show/7496/carrot.svg",
                },
              }
            }
            position={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <ForwardRef(Leaflet(Popup))>
              <h3 />
              <p />
              <p />
              <p />
            </ForwardRef(Leaflet(Popup))>
          </ForwardRef(Leaflet(Marker))>,
        ],
      ],
      "zoom": 11,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "attribution": "© <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors",
          "url": "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef(Leaflet(Popup))>
            <h3 />
            <p />
            <p />
            <p />
          </ForwardRef(Leaflet(Popup))>,
          "icon": NewClass {
            "_initHooksCalled": true,
            "options": Object {
              "iconUrl": "https://www.svgrepo.com/show/7496/carrot.svg",
            },
          },
          "position": Array [
            undefined,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <h3 />,
              <p />,
              <p />,
              <p />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "center": Array [
          39.750614,
          -104.996775,
        ],
        "children": Array [
          <ForwardRef(Leaflet(TileLayer))
            attribution="© <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors"
            url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
          />,
          Array [
            <ForwardRef(Leaflet(Marker))
              icon={
                NewClass {
                  "_initHooksCalled": true,
                  "options": Object {
                    "iconUrl": "https://www.svgrepo.com/show/7496/carrot.svg",
                  },
                }
              }
              position={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              <ForwardRef(Leaflet(Popup))>
                <h3 />
                <p />
                <p />
                <p />
              </ForwardRef(Leaflet(Popup))>
            </ForwardRef(Leaflet(Marker))>,
          ],
        ],
        "zoom": 11,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "attribution": "© <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors",
            "url": "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef(Leaflet(Popup))>
              <h3 />
              <p />
              <p />
              <p />
            </ForwardRef(Leaflet(Popup))>,
            "icon": NewClass {
              "_initHooksCalled": true,
              "options": Object {
                "iconUrl": "https://www.svgrepo.com/show/7496/carrot.svg",
              },
            },
            "position": Array [
              undefined,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <h3 />,
                <p />,
                <p />,
                <p />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Map should match snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Mapp
    vendorSearchResults={
      Array [
        Object {
          "attributes": Object {
            "account_type": "vendor",
            "address": "4950 Beach Court",
            "bio": "The WORLDS Freshest Basil",
            "city": "denver",
            "email": "helter_skelter@example.com",
            "img_url": null,
            "lat": 39.7861784,
            "long": -105.0178452,
            "name": "Basil Connection",
            "phone": 2313414141,
            "products": Array [
              Object {
                "Berries": Object {
                  "description": "lb of berries",
                  "id": 3,
                  "item_id": 1,
                  "price": 1400,
                  "unit": "lb",
                  "user_id": 5,
                },
              },
              Object {
                "Potatoes": Object {
                  "description": "a lb of potatoes",
                  "id": 17,
                  "item_id": 4,
                  "price": 170,
                  "unit": "lb",
                  "user_id": 5,
                },
              },
            ],
            "state": "CO",
            "zip": 80221,
          },
          "id": "5",
          "type": "user",
        },
      ]
    }
    vendors={
      Array [
        Object {
          "attributes": Object {
            "account_type": "vendor",
            "address": "4950 Beach Court",
            "bio": "The WORLDS Freshest Basil",
            "city": "denver",
            "email": "helter_skelter@example.com",
            "img_url": null,
            "lat": 39.7861784,
            "long": -105.0178452,
            "name": "Basil Connection",
            "phone": 2313414141,
            "products": Array [
              Object {
                "Berries": Object {
                  "description": "lb of berries",
                  "id": 3,
                  "item_id": 1,
                  "price": 1400,
                  "unit": "lb",
                  "user_id": 5,
                },
              },
              Object {
                "Potatoes": Object {
                  "description": "a lb of potatoes",
                  "id": 17,
                  "item_id": 4,
                  "price": 170,
                  "unit": "lb",
                  "user_id": 5,
                },
              },
            ],
            "state": "CO",
            "zip": 80221,
          },
          "id": "5",
          "type": "user",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "center": Array [
        39.750614,
        -104.996775,
      ],
      "children": Array [
        <ForwardRef(Leaflet(TileLayer))
          attribution="© <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors"
          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        />,
        Array [],
      ],
      "zoom": 11,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "attribution": "© <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors",
          "url": "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "center": Array [
          39.750614,
          -104.996775,
        ],
        "children": Array [
          <ForwardRef(Leaflet(TileLayer))
            attribution="© <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors"
            url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
          />,
          Array [],
        ],
        "zoom": 11,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "attribution": "© <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors",
            "url": "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
